### ISSUE TITLE
[Crash] Visual Glitches before crashing

### ISSUE THREAD

**Issue by GrandSteven** [Raytwo/Cobalt#31]
### System Firmware Version

16.1.0.

### Atmosphere Version

1.5.5.

### Cobalt Version

v0.30.2.

### Can the crash be consistently reproduced?

Yes

### At what moment does the crash happen?

- [ ] Before the Switch boot animation
- [ ] After the Switch boot animation
- [ ] While the picture is displayed on boot
- [ ] While the spinner is displayed on boot
- [x] At random
- [x] After a long time playing

### Details

![image](https://github.com/Raytwo/Cobalt/assets/146592283/5004a770-509d-4f2a-a8a2-709274fa40f7)
![IMG_0836](https://github.com/Raytwo/Cobalt/assets/146592283/0759bdce-9be0-4431-9400-ccccc60f068b)

After playing for a bit, game will show signs of error with visual glitches. after awhile, game would crash.

This happens with even no mods, as long as Cobalt is installed it seems to be crashing.

Game version is 2.0.0.

### Crash report

```shell
Atmosphère Crash Report (v1.7):
Result:                          0x0 (2000-0000)

Process Info:
    Process Name:                Application
    Program ID:                  0100a6301214e000
    Process ID:                  000000000000008b
    Process Flags:               00001877
    User Exception Address:      000000204c4bb000
Exception Info:
    Type:                        User Break
    Address:                     0000004bc9922644 (nnSdk + 0x397644)
    Break Reason:                0x0
    Break Address:               0000001100a3aef8
    Break Size:                  0x4
Crashed Thread Info:
    Thread ID:                   0000000000000331
    Thread Name:                 NxSystemThread
    Stack Region:                0000001100a3a000-0000001100a3b000
    Registers:
        X[00]:                   0000000000000000
        X[01]:                   0000001100a3aef8
        X[02]:                   0000000000000004
        X[03]:                   0000000000000000
        X[04]:                   0000000000000000
        X[05]:                   0000000000000000
        X[06]:                   0000000000000000
        X[07]:                   0000000000000000
        X[08]:                   0000000000000000
        X[09]:                   0000000000000000
        X[10]:                   0000000000000000
        X[11]:                   0000000000000000
        X[12]:                   0000000000000000
        X[13]:                   0000000000000000
        X[14]:                   0000000000000000
        X[15]:                   0000000000000000
        X[16]:                   0000000000000000
        X[17]:                   0000000000000000
        X[18]:                   0000000000000000
        X[19]:                   0000000000000000
        X[20]:                   0000000000000000
        X[21]:                   0000000000000000
        X[22]:                   0000000000000000
        X[23]:                   0000000000000000
        X[24]:                   0000000000000000
        X[25]:                   0000000000000000
        X[26]:                   0000000000000000
        X[27]:                   0000000000000000
        X[28]:                   0000000000000000
        FP:                      0000001100a3ade0
        LR:                      0000004bc96d3918 (nnSdk + 0x148918) (_ZN2nn4diag6detail5AbortEPKNS_6ResultE + 0x18)
        SP:                      0000001100a3ade0
        PC:                      0000004bc9922640 (nnSdk + 0x397640)
    Stack Trace:
        ReturnAddress[00]:       0000004bc96d20d8 (nnSdk + 0x1470d8)
        ReturnAddress[01]:       0000004bc96d2238 (nnSdk + 0x147238)
        ReturnAddress[02]:       0000004bc98c6828 (nnSdk + 0x33b828)
        ReturnAddress[03]:       0000004bc98cf314 (nnSdk + 0x344314)
        ReturnAddress[04]:       0000004bc98cf2c8 (nnSdk + 0x3442c8)
        ReturnAddress[05]:       0000004bc98cf2bc (nnSdk + 0x3442bc)
        ReturnAddress[06]:       0000004bc98cd188 (nnSdk + 0x342188)
        ReturnAddress[07]:       0000004bc98d3714 (nnSdk + 0x348714)
        ReturnAddress[08]:       0000000000000000
    Stack Dump:                               00 01 02 03 04 05 06 07 08 09 0a 0b 0c 0d 0e 0f
                                 001100a3ade0 70 ae a3 00 11 00 00 00 d8 20 6d c9 4b 00 00 00
                                 001100a3adf0 03 00 00 00 00 00 00 00 60 ae a3 00 11 00 00 00
                                 001100a3ae00 79 be fb c9 4b 00 00 00 79 be fb c9 4b 00 00 00
                                 001100a3ae10 79 be fb c9 4b 00 00 00 00 00 00 00 00 00 00 00
                                 001100a3ae20 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
                                 001100a3ae30 03 00 00 00 00 00 00 00 60 ae a3 00 11 00 00 00
                                 001100a3ae40 79 be fb c9 4b 00 00 00 79 be fb c9 4b 00 00 00
                                 001100a3ae50 79 be fb c9 4b 00 00 00 00 00 00 00 00 00 00 00
                                 001100a3ae60 79 be fb c9 4b 00 00 00 d8 ae a3 00 11 00 00 00
                                 001100a3ae70 00 af a3 00 11 00 00 00 38 22 6d c9 4b 00 00 00
                                 001100a3ae80 00 00 00 00 00 00 00 00 f8 c1 25 ca 4b 00 00 00
                                 001100a3ae90 00 00 00 00 00 00 00 00 f0 67 8c c9 4b 00 00 00
                                 001100a3aea0 50 e2 29 ca 4b 00 00 00 78 cb 29 ca 4b 00 00 00
                                 001100a3aeb0 f8 e1 29 ca 4b 00 00 00 28 e2 29 ca 4b 00 00 00
                                 001100a3aec0 c0 e1 29 ca 4b 00 00 00 c8 e1 29 ca 4b 00 00 00
                                 001100a3aed0 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
    TLS Address:                 000000204c4bb400
    TLS Dump:                                 00 01 02 03 04 05 06 07 08 09 0a 0b 0c 0d 0e 0f
                                 00204c4bb400 00 00 00 00 0c 00 00 00 00 00 00 00 00 00 00 00
                                 00204c4bb410 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
                                 00204c4bb420 53 46 43 4f 00 00 00 00 00 00 00 00 0e 61 f9 21
                                 00204c4bb430 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
                                 00204c4bb440 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
                                 00204c4bb450 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
                                 00204c4bb460 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
                                 00204c4bb470 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
                                 00204c4bb480 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
                                 00204c4bb490 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
                                 00204c4bb4a0 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
                                 00204c4bb4b0 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
                                 00204c4bb4c0 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
                                 00204c4bb4d0 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
                                 00204c4bb4e0 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
                                 00204c4bb4f0 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
Module Info:
    Number of Modules:           5
    Module 00:
        Address:                 0000004bc2600000-0000004bc2602000
        Name:                    nnrtld
        Module Id:               E79AB9AF2AB0517496B6C5630B5C7A69769806D4000000000000000000000000
    Module 01:
        Address:                 0000004bc2604000-0000004bc654c000
        Name:                    SwitchPlayer.nss
        Module Id:               8C08B9719E085F91847B5E0F935D948800000000000000000000000000000000
    Module 02:
        Address:                 0000004bc8cb3000-0000004bc8ff3000
        Name:                    multimedia
        Module Id:               BFFED55EA46C1B48DDEABCACDFBF81BF418125AC000000000000000000000000
    Module 03:
        Address:                 0000004bc934d000-0000004bc94ea000
        Name:                    cobalt
        Module Id:               0000000000000000000000000000000000000000000000000000000000000000
    Module 04:
        Address:                 0000004bc958b000-0000004bc9b1b000
        Name:                    nnSdk
        Module Id:               56EC24747EC9E7F19B3EFB062483A19D1DA091A8000000000000000000000000
```

**Comment by DogeThis** [comment]
Please update to 0.31.0 for now.

**Comment by DogeThis** [comment]
Thanks for the detailed report.

**Comment by Raytwo** [comment]
This was (hopefully) addressed with version 0.31.0. I'll be closing this for now, but if the issue is still present, feel free to reopen an issue! Thanks again for the report.
